indexfile_unix.go

```diff
@@ -29,6 +29,9 @@ type mmapedIndexFile struct {
 }
 
 func (f *mmapedIndexFile) Read(off, sz uint32) ([]byte, error) {
+	if off > off+sz {
```

Comment:
comment `// test for wrapped overflow`

maybe combine with the other bound test below?